PyPortal_Hurricane_Tracker/hurricane_tracker.py

```diff
@@ -0,0 +1,171 @@
+import time
+import math
+import board
+import displayio
+import terminalio
+from simpleio import map_range
+import adafruit_imageload
+from adafruit_pyportal import PyPortal
+from adafruit_display_text.label import Label
+from adafruit_display_shapes.line import Line
+
+# --| User Config |---------------------------------------------------
+UPDATE_RATE = 60  # minutes
+MAX_STORMS = 3  # limit storms
+NAME_COLOR = 0xFFFFFF  # label text color
+NAME_BG_COLOR = 0x000000  # label background color
+ARROW_COLOR = 0x0000FF  # movement direction arrow color
+ARROW_LENGTH = 15  # movement direction arrow length
+LAT_RANGE = (45, 5)  # set to match map
+LON_RANGE = (-100, -40)  # set to match map
+# --------------------------------------------------------------------
+
+# setup pyportal
+pyportal = PyPortal(
+    url="https://www.nhc.noaa.gov/CurrentStorms.json",
+    json_path=["activeStorms"],
+    status_neopixel=board.NEOPIXEL,
+    default_bg="/map.bmp",
+)
+
+# setup display group for storms
+icons_bmp, icons_pal = adafruit_imageload.load(
+    "/storm_icons.bmp", bitmap=displayio.Bitmap, palette=displayio.Palette
+)
+for i, c in enumerate(icons_pal):
+    if c == 0xFFFF00:
+        icons_pal.make_transparent(i)
+storm_icons = displayio.Group(max_size=MAX_STORMS)
+pyportal.splash.append(storm_icons)
+STORM_CLASS = ("TD", "TS", "HU")
+
+# setup info label
+info_update = Label(
+    terminalio.FONT,
+    text="1984-01-01T00:00:00.000Z",
+    color=NAME_COLOR,
+    background_color=NAME_BG_COLOR,
+)
+info_update.anchor_point = (0.0, 1.0)
+info_update.anchored_position = (10, board.DISPLAY.height - 10)
+pyportal.splash.append(info_update)
+
+# these are need for lat/lon to screen x/y mapping
+VIRTUAL_WIDTH = board.DISPLAY.width * 360 / (LON_RANGE[1] - LON_RANGE[0])
+VIRTUAL_HEIGHT = board.DISPLAY.height * 360 / (LAT_RANGE[0] - LAT_RANGE[1])
+Y_OFFSET = math.radians(LAT_RANGE[0])
+Y_OFFSET = math.tan(math.pi / 4 + Y_OFFSET / 2)
+Y_OFFSET = math.log(Y_OFFSET)
+Y_OFFSET = (VIRTUAL_WIDTH * Y_OFFSET) / (2 * math.pi)
+Y_OFFSET = VIRTUAL_HEIGHT / 2 - Y_OFFSET
+
+
+def update_display():
+    # clear out existing icons
+    while len(storm_icons):
+        _ = storm_icons.pop()
+
+    # get latest storm data
+    try:
+        storm_data = pyportal.fetch()
+    except RuntimeError:
+        return
+    print("Number of storms:", len(storm_data))
```

Comment:
This prints # of storms but later only storms in area are displayed

Reply:
In general, all the serial prints are debug info. For this one, was helpful to know actual storm count in raw JSON.

Reply:
This is fine.. and useful.